src/passes/stringify-walker-impl.h

```diff
@@ -31,26 +31,15 @@ inline void StringifyWalker<SubType>::doWalkModule(Module* module) {
 
 template<typename SubType>
 inline void StringifyWalker<SubType>::doWalkFunction(Function* func) {
+  addUniqueSymbol(SeparatorCtx::makeFuncStart(func));
   walk(func->body);
-  /*
-   * We add a unique symbol after walking the function body to separate the
-   * string generated from visiting the function body as a single unit from the
-   * subsequent strings that will be generated from visiting the sub-expressions
-   * of the function body. If we did not add this unique symbol and a program
-   * had two functions with the same instructions, we would incorrectly create a
-   * new function with the instructions repeated twice.
-   *
-   * It might be helpful to think of the function body as a block that needs to
-   * be separated from subsequent instructions.
-   */
-  addUniqueSymbol();
 }
 
 template<typename SubType>
 inline void StringifyWalker<SubType>::walk(Expression* curr) {
   Super::walk(curr);
+  addUniqueSymbol(SeparatorCtx::makeFuncEnd(this->getFunction()));
   do {
-    addUniqueSymbol();
```

Comment:
Unrelated to this change, but if we made this a while loop instead of a do-while loop, I think we would be able to get rid of the queue empty check in `dequeueControlFlow`.

Reply:
Great idea! done

Reply:
This looks great, thanks 🎉